source/developer/interactive-message-dialogs.rst

```diff
@@ -0,0 +1,272 @@
+
+Select Elements
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Select elements are message menus that allow users to select one pre-defined option from a list. Below is an example of a select element that asks for one of three different options.
+
+// XXX <add screenshot for an example>
+
+.. code-block:: json
+
+  {
+    "display_name": "Options",
+    "name": "options",
+    "type": "select",
+    "options": [
+       {
+         "text": "Option1",
+         "value": "opt1"
+       },
+       {
+         "text": "Option2",
+         "value": "opt2"
+       },
+       {
+         "text": "Option3",
+         "value": "opt3"
+       }
+     ]
+  }
+
+Note that the JSON payload for ``select`` type elements matches :doc:`interactive message menus <interactive-messages>`.
+
+The select element can also be generated dynamically from users and channels of the system.
+
+For users, use
+
+.. code-block:: json
+
+  {
+    "display_name": "Assignee",
+    "name": "assignee",
+    "type": "select",
+    "data_source": "users"
+  }
+
+// XXX <add screenshot for an example>
+
+and for public channels, use
+
+  {
+    "display_name": "Post this message to",
+    "name": "channel",
+    "type": "select",
+    "data_source": "channels"
+  }
+
+// XXX <add screenshot for an example>
+
+The list of supported fields for the ``select`` type element is included below:
+
+.. csv-table::
+    :header: "Field", "Type", "Description"
+
+    "display_name", "String", "Display name of the field shown to the user in the message dialog. Maximum 24 characters."
+    "name", "String", "Name of the field element used by the integration. Maximum 300 characters. You should use unique “name” fields in the same message dialog."
+    "type", "String", "Set this value to ``select`` for a select element."
+    "data_source", "String", "(Optional) One of ``users``, or ``channels``. If none specified, assumes a manual list of options is provided by the integration."
+    "options", "String", "(Optional) An array of options for the select element. Not applicable for ``users`` or ``channels`` data sources."
+    "optional", "String", "(Optional) Set to true if this form element is not required. Default is ``false``."
+    "help_text", "Integer", "(Optional) Set help text for this form element. Maximum 150 characters."
+    "default", "Integer", "(Optional) Set a default value for this form element. Maximum 3,000 characters."
+    "placeholder", "Integer", "(Optional) A string displayed to help guide users in completing the element. Maximum 3,000 characters."
+
+Dialog Submission
+-----------------------
+
+When a user submits a dialog, Mattermost will perform client-side input validation to make sure
+
+  - All required fields are filled
+  - All formats are correct (e.g. email, telephone number, etc.)
+
+Moreover, Mattermost also allows the integration itself to perform input validation.
+
+// XXX Joram can we have an example of this? Using a Go example is okay.
```

Comment:
Joram: This question is for you

Reply:
I don't think a code example would be the most use here. A JSON response example like all the other sections would make most sense. For the test payload I gave you, I returned the following as the response body to the request to submit the dialog if the number wasn't 42:

```json
{"errors": {"somenumber": "Number must be 42!"}}
```

`somenumber` is the name of the number field in my test payload

I would also suggest moving the input validation to be discussed after we discuss what the submission payload looks like, since that will be the order any integration has to deal with them.